### PR TITLE
Additional features for extractor + several fixes

### PR DESCRIPTION
**Pull request by johanneshiry**
This PR addresses #78 again and extends the functionality of the extractor.
Furthermore some bugfixes and improvements are included:

- added missing constructors for all thermal models (they have been out of operation by default and couldn't be turned on)
- fixes class names in UML for CylindricStorageInput
- fixes null assignment for an OperatorInput in AssetInput 
- fixes invalid javadoc in ThermalHouseInput
- adapted file naming strategy for easier conversion of camel case to snake case
- extended tests for CsvFileSink with new models 
- extended tests for Extractor with new models 
- fixed documentation for `OperationTime`
- changed constructor signature to harmonize their signature to `uuid, id, operator, operationTime`

### CODE REVIEW

File: src/main/java/edu/ie3/datamodel/io/FileNamingStrategy.java [hunk 1]
```diff
@@ -28,15 +29,11 @@
...
-  private static final String INPUT_ENTITY_SUFFIX = "_input";
-  private static final String TYPE_INPUT = "_type_input";
-  private static final String GRAPHIC_INPUT_SUFFIX = "_graphic";
-
-  private static final String INPUT_CLASS_STRING = "Input";
 
   private final String prefix;
   private final String suffix;
 
+  /** Constructor for building the file names without files with prefix and suffix */
```
Comment (sensarmad) [inline src/main/java/edu/ie3/datamodel/io/FileNamingStrategy.java:36]: without provided prefix and suffix

File: src/main/java/edu/ie3/datamodel/io/extractor/Extractor.java [hunk 2]
```diff
@@ -19,41 +19,51 @@
...
-public class Extractor {
+public final class Extractor {
 
-  private final List<InputEntity> extractedEntities;
+  private static final Logger log = LogManager.getLogger(Extractor.class);
 
-  public Extractor(Nested nestedEntity) throws ExtractorException {
-    this.extractedEntities = extractElements(nestedEntity);
+  private Extractor() {
+    throw new IllegalStateException("Don't instantiate a utility class");
```
Comment (sensarmad) [inline src/main/java/edu/ie3/datamodel/io/extractor/Extractor.java:27]: Our software should not order people around :D Please change to "Utility classes cannot be instantiated"

File: src/main/java/edu/ie3/datamodel/io/factory/input/NodeInputFactory.java [hunk 3]
```diff
@@ -48,6 +48,6 @@ protected NodeInput buildModel(
     final VoltageLevel voltLvl = data.getVoltageLvl(VOLT_LVL.toLowerCase(), V_RATED.toLowerCase());
     final int subnet = data.getInt(SUBNET);
     return new NodeInput(
-        uuid, operationTime, operatorInput, id, vTarget, slack, geoPosition, voltLvl, subnet);
+        uuid, operationTime, operator, id, vTarget, slack, geoPosition, voltLvl, subnet);
```
Comment (sensarmad) [inline src/main/java/edu/ie3/datamodel/io/factory/input/NodeInputFactory.java:51]: I belive you want to go with uuid, id, operator, operationTime(like above)

File: src/main/java/edu/ie3/datamodel/io/factory/input/Transformer2WInputFactory.java [hunk 4]
```diff
@@ -34,23 +34,14 @@ protected Transformer2WInput buildModel(
...
-        operationTime,
-        operatorInput,
-        id,
-        nodeA,
-        nodeB,
-        parallelDevices,
-        type,
-        tapPos,
-        autoTap);
+        uuid, operationTime, operator, id, nodeA, nodeB, parallelDevices, type, tapPos, autoTap);
```
Comment (sensarmad) [inline src/main/java/edu/ie3/datamodel/io/factory/input/Transformer2WInputFactory.java:45]: uuid, id, operator, operatorTime

File: src/main/java/edu/ie3/datamodel/models/OperationTime.java [hunk 5]
```diff
@@ -32,14 +32,10 @@ protected OperationTime(ZonedDateTime startDate, ZonedDateTime endDate, boolean
     this.isLimited = isLimited;
   }
 
-  /** Constructor for OperationTime without limitations */
+  /** Constructor for OperationTime without limitations (= always one) */
```
Comment (sensarmad) [inline src/main/java/edu/ie3/datamodel/models/OperationTime.java:35]: always on

File: src/main/java/edu/ie3/datamodel/models/input/thermal/CylindricalStorageInput.java [hunk 6]
```diff
@@ -26,6 +28,37 @@
...
+   * @param uuid Unique identifier of a cylindrical storage
+   * @param id Identifier of the thermal unit
+   * @param operator operator of the asset
+   * @param operationTime operation time of the asset
+   * @param bus Thermal bus, a thermal unit is connected to
+   * @param storageVolumeLvl Available storage volume
+   * @param storageVolumeLvlMin Minimum permissible storage volume
+   * @param inletTemp Temperature of the inlet
+   * @param returnTemp Temperature of the outlet
+   * @param c Specific heat capacity of the storage medium
```
Comment (sensarmad) [inline src/main/java/edu/ie3/datamodel/models/input/thermal/CylindricalStorageInput.java:41]: Nothing really necessary: But if the common parameters had common definitions and text format throughout...would be great :)

File: src/test/groovy/edu/ie3/datamodel/io/factory/input/AssetInputEntityFactoryTest.groovy [hunk 7]
```diff
@@ -279,8 +279,8 @@ class AssetInputEntityFactoryTest extends Specification implements FactoryTestHe
 		}
 
 		@Override
-		protected TestAssetInput buildModel(AssetInputEntityData data, UUID uuid, String id, OperatorInput operatorInput, OperationTime operationTime) {
-			return new TestAssetInput(uuid, operationTime, operatorInput, id)
+		protected TestAssetInput buildModel(AssetInputEntityData data, UUID uuid, String id, OperatorInput operator, OperationTime operationTime) {
+			return new TestAssetInput(uuid, operationTime, operator, id)
```
Comment (sensarmad) [inline src/test/groovy/edu/ie3/datamodel/io/factory/input/AssetInputEntityFactoryTest.groovy:283]: uuid, id, operator, operationTime?

File: src/test/groovy/edu/ie3/datamodel/io/factory/input/participant/EvInputFactoryTest.groovy [hunk 8]
```diff
@@ -41,7 +41,7 @@ class EvInputFactoryTest extends Specification implements FactoryTestHelper {
 
 		when:
 		Optional<EvInput> input = inputFactory.getEntity(
-				new SystemParticipantTypedEntityData<EvTypeInput>(parameter, inputClass, operatorInput, nodeInput, typeInput))
+				new SystemParticipantTypedEntityData<EvTypeInput>(parameter, inputClass,operatorInput, nodeInput, typeInput))
```
Comment (sensarmad) [inline src/test/groovy/edu/ie3/datamodel/io/factory/input/participant/EvInputFactoryTest.groovy:44]: :(

File: src/test/groovy/edu/ie3/datamodel/io/factory/input/participant/HpInputFactoryTest.groovy [hunk 9]
```diff
@@ -43,7 +43,7 @@ class HpInputFactoryTest extends Specification implements FactoryTestHelper {
 
 		when:
 		Optional<HpInput> input = inputFactory.getEntity(
-				new HpInputEntityData(parameter, operatorInput, nodeInput, typeInput, thermalBusInput))
+				new HpInputEntityData(parameter,operatorInput, nodeInput, typeInput, thermalBusInput))
```
Comment (sensarmad) [inline src/test/groovy/edu/ie3/datamodel/io/factory/input/participant/HpInputFactoryTest.groovy:46]: :(

File: src/main/java/edu/ie3/datamodel/io/extractor/HasNodes.java [hunk 10]
```diff
@@ -6,15 +6,15 @@
 package edu.ie3.datamodel.io.extractor;
 
 import edu.ie3.datamodel.models.input.NodeInput;
+import java.util.List;
 
 /**
- * Interface that should be implemented by all elements holding a {@link NodeInput} and should be
- * processable by the {@link Extractor}
+ * Interface that should be implemented by all elements holding a one or more {@link NodeInput}
```
Comment (sensarmad) [inline src/main/java/edu/ie3/datamodel/io/extractor/HasNodes.java:12]: holding one or more

File: src/test/groovy/edu/ie3/datamodel/io/factory/input/participant/EvInputFactoryTest.groovy (hunk 8 above)
Comment (johanneshiry) [reply src/test/groovy/edu/ie3/datamodel/io/factory/input/participant/EvInputFactoryTest.groovy:44]: @sensarmad don't get this - would you mind clarifying pls?!

File: src/test/groovy/edu/ie3/datamodel/io/factory/input/participant/HpInputFactoryTest.groovy (hunk 9 above)
Comment (johanneshiry) [reply src/test/groovy/edu/ie3/datamodel/io/factory/input/participant/HpInputFactoryTest.groovy:46]: @sensarmad don't get this - would you mind clarifying pls?!

**Pull request merged**